automation/roles/pgbouncer/tasks/main.yml

```diff
@@ -124,6 +124,27 @@
     label: "{{ 'pgbouncer' if idx == 0 else 'pgbouncer-%d' % (idx + 1) }}"
   tags: pgbouncer_logrotate, pgbouncer
 
+# Generating TLS Certificates
+- block:
+    - name: Generate a private key for TLS
+      become: true
+      become_user: postgres
+      community.crypto.openssl_privatekey:
+        path: "{{ pgbouncer_conf_dir }}/server.key"
+        size: '4096'
+        mode: '0400'
+
+    - name: Generate a self-signed certificate for PGBouncer
+      community.crypto.x509_certificate:
+        path: "{{ pgbouncer_conf_dir }}/server.crt"
+        privatekey_path: "{{ pgbouncer_conf_dir }}/server.key"
+        owner: postgres
+        group: postgres
+        mode: '0644'
+        provider: selfsigned
+        entrust_not_after: "+1000d"
+  when: pgbouncer_server_tls_sslmode | default('disable') != 'disable'
+
```

Comment:
@klention what do you think?

Reply:
Generating one pair of TLS cert sounds good. However, I moved the task for generating the cert/key to the PGBouncer role instead of PostgreSQL role since PGBouncer role executes first. This way PGBouncer tasks will not fail if the cert is not yet created.

Reply:
Maybe we should put this in a separate role? Considering that the TLS key and certificate will be used not only for pgbouncer.

Reply:
Agree. Created a new role named tls_certificate to host TLS generation tasks.
Made it dynamic through vars with parameters like the following example:

    tls_title: "PostgreSQL & PGBouncer" -> Will append this string to the name of the task for debugging purposes.
    tls_privatekey_path: "{{ pgbouncer_conf_dir }}/server.key" -> Full path of the private key location where it will be created.
    tls_cert_path: "{{ pgbouncer_conf_dir }}/server.crt" -> Full path of the certificate location where it will be created.
    tls_owner: "postgres" -> the owner and group for both key/cert

Reply:
I've made some small changes https://github.com/vitabaks/autobase/pull/849/commits/0aed35fbcef6919a78c8dac7aef538ef21bb9466